File: JobSearchCodeExamples/JobSearchCodeExamples.VB.Test/TestTaker/ToIntTest.vb
Imports JobSearchCodeExamples.VB.TestTaker
Imports Xunit

Namespace TestTaker

    Public Class ToIntTest

        ''' <summary>
        ''' Romans to int test.
        ''' </summary>
        <Theory>
        <MemberData(NameOf(ToIntTest.TestData))>
        Public Sub RomanToInt1Test(
                ByVal romanNumber As String,
                ByVal expectedResults As Integer)

            Assert.Equal(expectedResults, ToInt.RomanToInt(romanNumber))

        End Sub

        ''' <summary>
        ''' Romans to int test.
        ''' </summary>
        <Theory>
        <MemberData(NameOf(ToIntTest.TestData))>
        Public Sub RomanToInt2Test(
                ByVal romanNumber As String,
                ByVal expectedResults As Integer)

            Assert.Equal(expectedResults, ToInt.RomanToInt2(romanNumber))

        End Sub

        ''' <summary>
        ''' Romans to int test.
        ''' </summary>
        <Theory>
        <MemberData(NameOf(ToIntTest.TestData))>
        Public Sub RomanToInt3Test(
                ByVal romanNumber As String,
                ByVal expectedResults As Integer)

            Assert.Equal(expectedResults, ToInt.RomanToInt3(romanNumber))

        End Sub

        ''' <summary>
        ''' Gets the test data.
        ''' </summary>
        ''' <value>
        ''' The test data.
        ''' </value>
        Public Shared ReadOnly Property TestData As IEnumerable(Of Object())

            Get

                Dim result = New List(Of Object()) From {
                    New Object() {"I", 1},
                    New Object() {"II", 2},
                    New Object() {"III", 3},
                    New Object() {"IV", 4},
                    New Object() {"V", 5},
                    New Object() {"VI", 6},
                    New Object() {"VII", 7},
                    New Object() {"VIII", 8},
                    New Object() {"IX", 9},
                    New Object() {"X", 10},
                    New Object() {"XI", 11},
                    New Object() {"XX", 20},
                    New Object() {"XXI", 21},
                    New Object() {"XXII", 22},
                    New Object() {"XXIII", 23},
                    New Object() {"XXIV", 24},
                    New Object() {"XXV", 25},
                    New Object() {"XLV", 45},
                    New Object() {"LVIII", 58},
                    New Object() {"DCCC", 800},
                    New Object() {"MCMLXIII", 1963},
                    New Object() {"MCMXCIV", 1994},
                    New Object() {"MMC", 2100},
                    New Object() {"MMCD", 2400},
                    New Object() {"MMCM", 2900},
                    New Object() {"MMM", 3000},
                    New Object() {"MMMCMLXIV", 3964},
                    New Object() {"MMMCMLXV", 3965},
                    New Object() {"MMMCMLXVI", 3966},
                    New Object() {"MMMCMLXVII", 3967},
                    New Object() {"MMMCMLXVIII", 3968},
                    New Object() {"MMMCMLXIX", 3969},
                    New Object() {"MMMCMLXX", 3970},
                    New Object() {"MMMCMLXXI", 3971},
                    New Object() {"MMMCMLXXII", 3972},
                    New Object() {"MMMCMLXXIII", 3973},
                    New Object() {"MMMCMLXXIV", 3974},
                    New Object() {"MMMCMLXXV", 3975},
                    New Object() {"MMMCMLXXVI", 3976},
                    New Object() {"MMMCMLXXVII", 3977},
                    New Object() {"MMMCMLXXVIII", 3978},
                    New Object() {"MMMCMLXXIX", 3979},
                    New Object() {"MMMCMLXXX", 3980},
                    New Object() {"MMMCMLXXXI", 3981},
                    New Object() {"MMMCMLXXXII", 3982},
                    New Object() {"MMMCMLXXXIII", 3983},
                    New Object() {"MMMCMLXXXIV", 3984},
                    New Object() {"MMMCMLXXXV", 3985},
                    New Object() {"MMMCMLXXXVI", 3986},
                    New Object() {"MMMCMLXXXVII", 3987},
                    New Object() {"MMMCMLXXXVIII", 3988},
                    New Object() {"MMMCMLXXXIX", 3989},
                    New Object() {"MMMCMXC", 3990},
                    New Object() {"MMMCMXCI", 3991},
                    New Object() {"MMMCMXCII", 3992},
                    New Object() {"MMMCMXCIII", 3993},
                    New Object() {"MMMCMXCIV", 3994},
                    New Object() {"MMMCMXCV", 3995},
                    New Object() {"MMMCMXCVI", 3996},
                    New Object() {"MMMCMXCVII", 3997},
                    New Object() {"MMMCMXCVIII", 3998},
                    New Object() {"MMMCMXCIX", 3999}
                    }

                Return result
            End Get

        End Property

    End Class

End Namespace

File: JobSearchCodeExamples/JobSearchCodeExamples.Test/TestTaker/ToIntTest.cs
using JobSearchCodeExamples.TestTaker;

namespace JobSearchCodeExamples.Test.TestTaker;

public class ToIntTest
{
    /// <summary>
    /// Romans to int test.
    /// </summary>
    [Theory]
    [MemberData(nameof(TestData))]
    public void RomanToInt1Test(string romanNumber, int expectedResults)
    {
        Assert.Equal(expectedResults, ToInt.RomanToInt(romanNumber));
    }

    /// <summary>
    /// Romans to int test.
    /// </summary>
    [Theory]
    [MemberData(nameof(TestData))]
    public void RomanToInt2Test(string romanNumber, int expectedResults)
    {
        Assert.Equal(expectedResults, ToInt.RomanToInt2(romanNumber));
    }

    /// <summary>
    /// Romans to int test.
    /// </summary>
    [Theory]
    [MemberData(nameof(TestData))]
    public void RomanToInt3Test(string romanNumber, int expectedResults)
    {
        Assert.Equal(expectedResults, ToInt.RomanToInt3(romanNumber));
    }

    /// <summary>
    /// Gets the test data.
    /// </summary>
    /// <value>
    /// The test data.
    /// </value>
    public static IEnumerable<object[]> TestData
    {
        get
        {
            var result = new List<object[]>
            {
                new object[] { "I", 1 },
                new object[] { "II", 2 },
                new object[] { "III", 3 },
                new object[] { "IV", 4 },
                new object[] { "V", 5 },
                new object[] { "VI", 6 },
                new object[] { "VII", 7 },
                new object[] { "VIII", 8 },
                new object[] { "IX", 9 },
                new object[] { "X", 10 },
                new object[] { "XI", 11 },
                new object[] { "XX", 20 },
                new object[] { "XXI", 21 },
                new object[] { "XXII", 22 },
                new object[] { "XXIII", 23 },
                new object[] { "XXIV", 24 },
                new object[] { "XXV", 25 },
                new object[] { "XLV", 45 },
                new object[] { "LVIII", 58 },
                new object[] { "DCCC", 800 },
                new object[] { "MCMLXIII", 1963 },
                new object[] { "MCMXCIV", 1994 },
                new object[] { "MMC", 2100 },
                new object[] { "MMCD", 2400 },
                new object[] { "MMCM", 2900 },
                new object[] { "MMM", 3000 },
                new object[] { "MMMCMLXIV", 3964 },
                new object[] { "MMMCMLXV", 3965 },
                new object[] { "MMMCMLXVI", 3966 },
                new object[] { "MMMCMLXVII", 3967 },
                new object[] { "MMMCMLXVIII", 3968 },
                new object[] { "MMMCMLXIX", 3969 },
                new object[] { "MMMCMLXX", 3970 },
                new object[] { "MMMCMLXXI", 3971 },
                new object[] { "MMMCMLXXII", 3972 },
                new object[] { "MMMCMLXXIII", 3973 },
                new object[] { "MMMCMLXXIV", 3974 },
                new object[] { "MMMCMLXXV", 3975 },
                new object[] { "MMMCMLXXVI", 3976 },
                new object[] { "MMMCMLXXVII", 3977 },
                new object[] { "MMMCMLXXVIII", 3978 },
                new object[] { "MMMCMLXXIX", 3979 },
                new object[] { "MMMCMLXXX", 3980 },
                new object[] { "MMMCMLXXXI", 3981 },
                new object[] { "MMMCMLXXXII", 3982 },
                new object[] { "MMMCMLXXXIII", 3983 },
                new object[] { "MMMCMLXXXIV", 3984 },
                new object[] { "MMMCMLXXXV", 3985 },
                new object[] { "MMMCMLXXXVI", 3986 },
                new object[] { "MMMCMLXXXVII", 3987 },
                new object[] { "MMMCMLXXXVIII", 3988 },
                new object[] { "MMMCMLXXXIX", 3989 },
                new object[] { "MMMCMXC", 3990 },
                new object[] { "MMMCMXCI", 3991 },
                new object[] { "MMMCMXCII", 3992 },
                new object[] { "MMMCMXCIII", 3993 },
                new object[] { "MMMCMXCIV", 3994 },
                new object[] { "MMMCMXCV", 3995 },
                new object[] { "MMMCMXCVI", 3996 },
                new object[] { "MMMCMXCVII", 3997 },
                new object[] { "MMMCMXCVIII", 3998 },
                new object[] { "MMMCMXCIX", 3999 }
            };

            return result;
        }
    }
}
